### ISSUE TITLE
pip install pycares doesn't work as expected from pypy

### ISSUE THREAD

**Issue by thestick613** [saghul/pycares#36]
I'm on
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"

apt-add-repository ppa:pypy/ppa
apt-get update
apt-get install libffi-dev python-dev pypy pypy-dev

pypy -V
Python 2.7.10 (5.4.1+dfsg-1~ppa1~ubuntu16.04, Sep 06 2016, 23:12:02)
[PyPy 5.4.1 with GCC 5.4.0 20160609]

virtualenv -p pypy venv_pypy
source ~/venv_pypy/bin/activate

```
And 
```
pip install pycares

pip install pycares
Downloading/unpacking pycares
  Downloading pycares-2.1.0.tar.gz (226kB): 226kB downloaded
  Running setup.py (path:/root/venv_pypy/build/pycares/setup.py) egg_info for package pycares

    warning: no previously-included files matching '__pycache__' found under directory '*'
    warning: no previously-included files matching '*.py[co]' found under directory '*'
    warning: no previously-included files matching '*.[ao]' found under directory '*'
    warning: no previously-included files matching '*.so' found under directory '*'
    no previously-included directories found matching 'docs/_build'
Requirement already satisfied (use --upgrade to upgrade): cffi>=1.5.0 in /usr/lib/pypy/lib_pypy (from pycares)
Installing collected packages: pycares
  Running setup.py install for pycares
    generating cffi module 'build/temp.linux-x86_64-2.7/_pycares_cffi.c'
    Traceback (most recent call last):
      File "<module>", line 1, in <module>
      File "/root/venv_pypy/build/pycares/setup.py", line 55, in <module>
        **kwargs
      File "/usr/lib/pypy/lib-python/2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/root/venv_pypy/site-packages/setuptools/command/install.py", line 54, in run
        return _install.run(self)
      File "/usr/lib/pypy/lib-python/2.7/distutils/command/install.py", line 613, in run
        self.run_command('build')
      File "/usr/lib/pypy/lib-python/2.7/distutils/cmd.py", line 334, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/pypy/lib-python/2.7/distutils/command/build.py", line 127, in run
        self.run_command(cmd_name)
      File "/usr/lib/pypy/lib-python/2.7/distutils/cmd.py", line 334, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/pypy/lib-python/2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/pypy/lib_pypy/cffi/setuptools_ext.py", line 128, in run
        ext.sources[0] = make_mod(self.build_temp, pre_run)
      File "/usr/lib/pypy/lib_pypy/cffi/setuptools_ext.py", line 114, in make_mod
        updated = recompiler.make_c_source(ffi, module_name, source, c_file)
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 1365, in make_c_source
        verbose)
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 1340, in _make_c_or_py_source
        recompiler.collect_step_tables()
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 233, in collect_step_tables
        self._generate("ctx")
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 218, in _generate
        method(tp, realname)
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 962, in _generate_cpy_struct_ctx
        self._struct_ctx(tp, *self._struct_names(tp))
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 866, in _struct_ctx
        "field '%s.%s'" % (tp.name, fldname))
      File "/usr/lib/pypy/lib_pypy/cffi/recompiler.py", line 921, in _check_not_opaque
        "%s is of an opaque type (not declared in cdef())" % location)
    TypeError: struct sockaddr_in6: field 'sockaddr_in6.sin6_addr' is of an opaque type (not declared in cdef())
    Complete output from command /root/venv_pypy/bin/pypy -c "import setuptools, tokenize;__file__='/root/venv_pypy/build/pycares/setup.py';exec(compile(getattr(tokenize, 'open',:
```

This works tho:
```
git clone https://github.com/boytm/pycares.git
cd pycares
git checkout master
./build_inplace
python setup.py install
```

**Comment by saghul** [comment]
Thanks!

> This works tho:

That was the original source for the CFFI port, it has since been integrated and IIRC some platforms required changes in that specific are you run into trouble.  I'll take a peek as soon as I can.

**Comment by thestick613** [comment]
Thank you for your help and for your great library!

**Comment by saghul** [comment]
Quick heads up, I just published 2.1.1 with a fix for this bug.
